### ISSUE TITLE
/usr/bin/machine-config-daemon: no such file or directory

### ISSUE THREAD

**Issue by wking** [openshift/machine-config-operator#176]
Logs from a recent master:

```console
[core@wking-master-0 ~]$ journalctl
...
Nov 14 22:48:26 wking-master-0 hyperkube[802]: I1114 22:48:26.759484     802 reconciler.go:207] operationExecutor.VerifyControllerAttachedVolume started for volume "machine-config-daemon-token-m6sb8" (UniqueName
: "kubernetes.io/secret/660e568b-e85f-11e8-8f39-2282d965a3a6-machine-config-daemon-token-m6sb8") pod "machine-config-daemon-c8jnt" (UID: "660e568b-e85f-11e8-8f39-2282d965a3a6")
Nov 14 22:48:26 wking-master-0 systemd[1]: Started Kubernetes transient mount for /var/lib/kubelet/pods/660e568b-e85f-11e8-8f39-2282d965a3a6/volumes/kubernetes.io~secret/machine-config-daemon-token-m6sb8.
Nov 14 22:48:26 wking-master-0 systemd[1]: Starting Kubernetes transient mount for /var/lib/kubelet/pods/660e568b-e85f-11e8-8f39-2282d965a3a6/volumes/kubernetes.io~secret/machine-config-daemon-token-m6sb8.
Nov 14 22:48:27 wking-master-0 systemd[1]: Started crio-conmon-89cb5a3db42bbbef7217b5a16c0af0dde19ef85f1f679d38442cd458355043f5.scope.
Nov 14 22:48:27 wking-master-0 systemd[1]: Starting crio-conmon-89cb5a3db42bbbef7217b5a16c0af0dde19ef85f1f679d38442cd458355043f5.scope.
Nov 14 22:48:27 wking-master-0 systemd[1]: Scope libcontainer-7068-systemd-test-default-dependencies.scope has no PIDs. Refusing.
Nov 14 22:48:27 wking-master-0 systemd[1]: Scope libcontainer-7068-systemd-test-default-dependencies.scope has no PIDs. Refusing.
Nov 14 22:48:27 wking-master-0 systemd[1]: Created slice libcontainer_7068_systemd_test_default.slice.
--
Nov 14 22:48:33 wking-master-0 systemd[1]: Stopping libcontainer_7123_systemd_test_default.slice.
Nov 14 22:48:33 wking-master-0 systemd[1]: Started libcontainer container d7762af5c0b21aed261a775866671947a26ab5a0d26721805f25959500639c38.
Nov 14 22:48:33 wking-master-0 systemd[1]: Starting libcontainer container d7762af5c0b21aed261a775866671947a26ab5a0d26721805f25959500639c38.
Nov 14 22:48:33 wking-master-0 systemd[1]: Stopped libcontainer container d7762af5c0b21aed261a775866671947a26ab5a0d26721805f25959500639c38.
Nov 14 22:48:33 wking-master-0 systemd[1]: Stopping libcontainer container d7762af5c0b21aed261a775866671947a26ab5a0d26721805f25959500639c38.
Nov 14 22:48:33 wking-master-0 crio[831]: time="2018-11-14 22:48:33.759095707Z" level=error msg="Container creation error: container_linux.go:336: starting container process caused "exec: \"/usr/bin/machine-conf
ig-daemon\": stat /usr/bin/machine-config-daemon: no such file or directory"
Nov 14 22:48:33 wking-master-0 crio[831]: "
Nov 14 22:48:33 wking-master-0 hyperkube[802]: E1114 22:48:33.849383     802 remote_runtime.go:187] CreateContainer in sandbox "89cb5a3db42bbbef7217b5a16c0af0dde19ef85f1f679d38442cd458355043f5" from runtime serv
ice failed: rpc error: code = Unknown desc = container create failed: container_linux.go:336: starting container process caused "exec: \"/usr/bin/machine-config-daemon\": stat /usr/bin/machine-config-daemon: no 
such file or directory"
Nov 14 22:48:33 wking-master-0 hyperkube[802]: E1114 22:48:33.850090     802 kuberuntime_manager.go:737] container start failed: CreateContainerError: container create failed: container_linux.go:336: starting co
ntainer process caused "exec: \"/usr/bin/machine-config-daemon\": stat /usr/bin/machine-config-daemon: no such file or directory"
Nov 14 22:48:33 wking-master-0 hyperkube[802]: E1114 22:48:33.850142     802 pod_workers.go:186] Error syncing pod 660e568b-e85f-11e8-8f39-2282d965a3a6 ("machine-config-daemon-c8jnt_openshift-machine-config-operator(660e568b-e85f-11e8-8f39-2282d965a3a6)"), skipping: failed to "StartContainer" for "machine-config-daemon" with CreateContainerError: "container create failed: container_linux.go:336: starting container process caused \"exec: \\\"/usr/bin/machine-config-daemon\\\": stat /usr/bin/machine-config-daemon: no such file or directory\"\n"
Nov 14 22:48:34 wking-master-0 hyperkube[802]: I1114 22:48:34.667320     802 kuberuntime_manager.go:517] Container {Name:machine-config-daemon Image:registry.svc.ci.openshift.org/openshift/origin-v4.0-20181114203912@sha256:a058885f3294ff33962322389e6ca2e0cc8dfd916206503c837c5c2e1bbf68a5 Command:[] Args:[start] WorkingDir: Ports:[] EnvFrom:[] Env:[{Name:NODE_NAME Value: ValueFrom:&EnvVarSource{FieldRef:&ObjectFieldSelector{APIVersion:v1,FieldPath:spec.nodeName,},ResourceFieldRef:nil,ConfigMapKeyRef:nil,SecretKeyRef:nil,}}] Resources:{Limits:map[] Requests:map[]} VolumeMounts:[{Name:rootfs ReadOnly:false MountPath:/rootfs SubPath: MountPropagation:<nil>} {Name:var-run-dbus ReadOnly:false MountPath:/var/run/dbus SubPath: MountPropagation:<nil>} {Name:run-systemd ReadOnly:false MountPath:/run/systemd SubPath: MountPropagation:<nil>} {Name:etc-ssl-certs ReadOnly:true MountPath:/etc/ssl/certs SubPath: MountPropagation:<nil>} {Name:etc-mcd ReadOnly:true MountPath:/etc/machine-config-daemon SubPath: MountPropagation:<nil>} {Name:machine-config-daemon-token-m6sb8 ReadOnly:true MountPath:/var/run/secrets/kubernetes.io/serviceaccount SubPath: MountPropagation:<nil>}] VolumeDevices:[] LivenessProbe:nil ReadinessProbe:nil Lifecycle:nil TerminationMessagePath:/dev/termination-log TerminationMessagePolicy:File ImagePullPolicy:IfNotPresent SecurityContext:&SecurityContext{Capabilities:nil,Privileged:*true,SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,ReadOnlyRootFilesystem:nil,AllowPrivilegeEscalation:nil,RunAsGroup:nil,} Stdin:false StdinOnce:false TTY:false} is dead, but RestartPolicy says that we should restart it.
...
```

Commit for that image:

```console
$ oc --config=${INSTALL_DIR}/auth/kubeconfig adm release info registry.svc.ci.openshift.org/openshift/origin-release:v4.0-20181114203912 --commits | grep machine-config
  machine-config-controller                     https://github.com/openshift/machine-config-operator                       c39289b736b6ca0ce8376bbc26c245c22dc2cdce
  machine-config-daemon                         https://github.com/openshift/machine-config-operator                       c39289b736b6ca0ce8376bbc26c245c22dc2cdce
  machine-config-operator                       https://github.com/openshift/machine-config-operator                       c39289b736b6ca0ce8376bbc26c245c22dc2cdce
  machine-config-server                         https://github.com/openshift/machine-config-operator                       c39289b736b6ca0ce8376bbc26c245c22dc2cdce
```

Maybe this was what #174 was fixing, @abhinavdahiya?

**Comment by abhinavdahiya** [comment]
Yes, shuld be fixed by #174

**Comment by smarterclayton** [comment]
https://github.com/openshift/origin/pull/21496 will catch crash looping pods on a cluster.
